rllib/utils/tests/test_nested_dict.py

```diff
@@ -30,32 +34,37 @@ def set_invalid_item_2():
             ("b", "d"),
             ("c", "e", "f"),
             ("d", "g", "h", "i"),
+            ("j", "k"),
+            ("l",),
         ]
 
-        desired_values = [100, 200, 300, 400, 500]
+        # We have 5 leafs that are not empty and two empty leafs
+        desired_values = [100, 200, 300, 400, 500, NestedDict({}), NestedDict({})]
 
         foo_dict["aa"] = 100
         foo_dict["b", "c"] = 200
         foo_dict[("b", "d")] = 300
         foo_dict["c", "e"] = {"f": 400}
 
-        # Note: key ("c", "f") should not be a valid key since it is empty
-        foo_dict["c", "f"] = NestedDict()
-
         # test __len__
-        self.assertEqual(len(foo_dict), len(desired_keys) - 1)
+        # We have not yet included d, j and l in foo_dict
+        self.assertEqual(len(foo_dict), len(desired_keys) - 3)
 
         # test __iter__
-        self.assertEqual(list(iter(foo_dict)), desired_keys[:-1])
+        self.assertEqual(list(iter(foo_dict)), desired_keys[:-3])
 
         # this call will use __len__ and __iter__
-        foo_dict["d"] = {"g": NestedDict([(("h"), NestedDict({"i": 500}))])}
+        foo_dict["d"] = {"g": NestedDict([("h", NestedDict({"i": 500}))])}
+        foo_dict["j"] = {"k": {}}
+        foo_dict["l"] = {}
 
         # test asdict
         check(foo_dict.asdict(), desired_dict)
 
         # test __len__ again
-        self.assertEqual(len(foo_dict), len(desired_keys))
+        # We have included d, j and l in foo_dict, but j and l don't contribute to
+        # the length
```

Comment:
Can you add why they don't contribute to the length? because they are empty sub-roots of the tree structure with no leaves.